File: ecommerce-frontend/src/services/OrderService.js
import api from "./api";

const placeOrder = (data) => {
    return api.post("/orders", data);
};

const getMyOrders = () => {
    return api.get("/orders/my-orders");
};

const getOrderById = (id) => {
    return api.get(`/orders/${id}`);
};

const cancelOrder = (id) => {
    return api.put(`/orders/${id}/cancel`);
};

const downloadInvoice = (id) => {
    return api.get(`/orders/${id}/invoice`, {
        responseType: "blob"
    });
};

const trackOrder = (id) => {
    return api.get(`/orders/${id}/track`);
};

export default {
    placeOrder,
    getMyOrders,
    getOrderById,
    cancelOrder,
    downloadInvoice,
    trackOrder
};

File: ecommerce-backend/src/main/java/com/ecommerce/service/OrderService.java
package com.ecommerce.service;

import com.ecommerce.dto.OrderResponse;
import com.ecommerce.dto.PlaceOrderRequest;
import com.ecommerce.entity.OrderStatus;
import com.ecommerce.dto.OrderTrackingResponse;

import java.util.List;

public interface OrderService {

    // ============================================
    // USER APIs
    // ============================================

    /**
     * Place a new order from the current user's cart.
     */
    OrderResponse placeOrder(PlaceOrderRequest request);

    /**
     * Get all orders of the logged-in user.
     */
    List<OrderResponse> getMyOrders();

    OrderTrackingResponse trackOrder(Long orderId);

    /**
     * Get a specific order by ID.
     */
    OrderResponse getOrderById(Long orderId);

    /**
     * Cancel an order.
     */
    String cancelOrder(Long orderId);

    // ============================================
    // ADMIN APIs
    // ============================================

    /**
     * Get all orders.
     */
    List<OrderResponse> getAllOrders();

    /**
     * Update order status.
     */
    OrderResponse updateOrderStatus(
            Long orderId,
            OrderStatus orderStatus
    );

}